atlp-rwanda/Phantom-frontend-pirates
# bus simulation

### What does this PR do?
This is the feature of simulating the bus in Phantom web application.
### Description of Task to be completed?
- ability to control bus by starting, stoping, moving the bus.
- adding passengers and removing them.
- user can see bus status in real-time.
- appearance of Notification on performed action

### How should this be manually tested?
1. To test it you navigate to **/simulation** link
2. To test the simulator, You have to turn on the bus,  and press the button to start/ignite which will start the bus.
3. You can start to move the bus and stop it.
4. You can add passengers and remove them when the car is stopped.
5. If you do anything unusual and usual you will get notifications about what you did or you have to do.

For user map:
1. The New route **/user-map** contains Location 
2. You have to select a Source and destination
3. Source and destination can't be the same.
4. You can see the Plot of the route

#### Any background context you want to provide?
This PR is being enhanced to show the movement of bus in real time on map.

### Preview
<img src="https://i.ibb.co/YWrvpbm/1.png" alt="1" border="0" />
<img src="https://i.ibb.co/Fkrpczh/usermap.png" alt="usermap" border="0" />
